File: Diagrammatize.Design.Pattern/code/cxx/FactoryMethod/A2/idcard/IDCard.cpp
#include "IDCard.hpp"
#include <iostream>

IDCard::IDCard(const std::string& owner_, int serial_): owner(owner_), serial(serial_) {
    std::cout << "制作" << owner << "(" << serial << ")" << "的ID卡。" << std::endl;
}

void IDCard::use() {
    std::cout << "使用" << owner << "(" << serial << ")" << "的ID卡。" << std::endl;
}

std::string IDCard::getOwner() {
    return owner;
}

int IDCard::getSerial() {
    return serial;
}

File: Diagrammatize.Design.Pattern/code/python/FactoryMethod/A2/idcard/IDCard.py
from framework import Product

class IDCard(Product):
    def __init__(self, owner, serial):
        print("制作" + owner + "(" + str(serial) + ")" + "的ID卡。")
        self.owner = owner
        self.serial = serial

    def use(self):
        print("使用" + self.owner + "(" + str(self.serial) + ")" + "的ID卡。")

    def getOwner(self):
        return self.owner

    def getSerial(self):
        return self.serial
